### PR TITLE
Adding sync to master in format for JSON and YMLs

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/30577

## Description
Currently, when a developer creates a PR after updating existing pack, github may show huge difference between the files (JSON or YML) because of their order.
I added a feature to sync the pack with the content's master. It will format the file according to the content's master structure and order.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: demisto_sdk/commands/format/README.md [hunk 1]
```diff
@@ -43,6 +43,10 @@ When done formatting, the **validate** command will run, to let you know of thin
 
   Set if you want to deprecate the integration/script/playbook
 
+* **-sm, --sync-to-master**
+
+  Set if you want to sync the yml file to content's master structure
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/format/README.md:48]:
```suggestion
  Set if you want to sync the yml file to XSOAR content's master structure
```

**Review by ShahafBenYakir — COMMENTED** [review]
Very nice

File: demisto_sdk/commands/format/update_generic.py [hunk 2]
```diff
@@ -370,3 +373,9 @@ def initiate_file_validator(self) -> int:
 
             else:
                 return SUCCESS_RETURN_CODE
+
+    def sync_data_to_master(self):
+        data_from_master = get_remote_file(self.relative_content_path, github_repo=GitContentConfig.OFFICIAL_CONTENT_REPO_NAME)
```
Comment (DeanArbel) [inline demisto_sdk/commands/format/update_generic.py:378]: In the contribution flow `prev_ver='xsoar-contrib/master'`.
@ShahafBenYakir wdyt? should `content/master` be ok, or should it be configurable?

**Review by DeanArbel — COMMENTED** [review]
Looks good to me, but I'd like to make sure that using `content/master` as a base covers all of our usecases.
@ShahafBenYakir Please review this as well.

File: demisto_sdk/commands/format/update_generic.py (hunk 2 above)
Comment (ShahafBenYakir) [reply demisto_sdk/commands/format/update_generic.py:378]: I think it's ok as:
a. The command could potentially run on other repos.
b. The PR opened is between the new branch and **content master** so it make more sense for it be identical to it.

File: demisto_sdk/commands/format/tests/test_formatting_yml_test.py [hunk 3]
```diff
@@ -1069,3 +1070,64 @@ def test_remove_spaces_end_of_id_and_name(self, pack, mocker, name):
         base_update_yml = BaseUpdateYML(input=integration.yml.path)
         base_update_yml.remove_spaces_end_of_id_and_name()
         assert base_update_yml.data['name'] == 'MyIntegration'
+
+    def test_sync_to_maser_no_change(self, mocker, tmp_path):
```
Comment (DeanArbel) [inline demisto_sdk/commands/format/tests/test_formatting_yml_test.py:1074]: please fix the typo. master.

File: demisto_sdk/commands/init/contribution_converter.py [hunk 4]
```diff
@@ -263,7 +263,8 @@ def format_converted_pack(self) -> None:
             update_docker=True,
             verbose=True,
             assume_yes=True,
-            prev_ver='xsoar-contrib/master'  # default is demisto/master
+            prev_ver='xsoar-contrib/master',  # default is demisto/master
+            sync_to_master=True
```
Comment (DeanArbel) [inline demisto_sdk/commands/init/contribution_converter.py:267]: please remove

**Review by DeanArbel — APPROVED** [review]
Please fix the comments below.

**Pull request merged**